### PR TITLE
Charts panel open/close toggle

### PR DESCRIPTION
**Pull request by uniphil**

### CODE REVIEW

File: app/scripts/components/dashboard/charts-container.jsx [hunk 1]
```diff
@@ -7,16 +7,30 @@ require('stylesheets/dashboard/charts-container');
...
+      <div className={`charts-container ${stateClass}`}>
+        <div className="above">
+          <div className="charts-container-header">
+            <span
+                className="tab-label"
+                onClick={this.props.onToggle}
+                role="button">
+              <T k="charts.toggle.activate" />
+            </span>
+            <ViewMode />
```
Comment (esoliani) [inline app/scripts/components/dashboard/charts-container.jsx:29]: ViewMode should be ouside of the charts-containter-header, shouldn't it?
Let's leave it and change it on the future

File: app/scripts/stores/layout.es6 [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  initialData: {
+    charts: charts.Closed(),
+  },
+  mixins: [SaneStore],
+  init() {
+    this.listenTo(toggleCharts, 'toggleCharts');
+  },
+  update(what, to) {
+    this.setData({
+      ...this.get(),
```
Comment (esoliani) [inline app/scripts/stores/layout.es6:17]: I'm not used to this notation, and it means different on es6 and jsx

File: app/scripts/components/dashboard/waterpoints.jsx [hunk 3]
```diff
@@ -40,13 +43,12 @@ const WaterPoints = React.createClass({
               retry={load}
               state={this.state.waterpointsState} />
         </div>
-        <ChartsContainer>
-          charts for waterpoints...
+        <ChartsContainer
+            onToggle={toggleCharts}
+            state={this.state.layout.charts}>
           <MetricStatus metric="54.65" title="chart.title.functional" total="123456"/>
```
Comment (esoliani) [inline app/scripts/components/dashboard/waterpoints.jsx]: I think these should be always visible.
In here we should add new component with chats

File: app/scripts/components/dashboard/charts-container.jsx (hunk 1 above)
Comment (uniphil) [inline app/scripts/components/dashboard/charts-container.jsx:29]: yeah hard to name things... `header` was meant as the nav-bar at the top of charts-container, so maybe it should be `charts-container-nav`?

File: app/scripts/components/dashboard/waterpoints.jsx (hunk 3 above)
Comment (uniphil) [inline app/scripts/components/dashboard/waterpoints.jsx]: yep you're right. I'll fix it.

File: app/scripts/stores/layout.es6 (hunk 2 above)
Comment (uniphil) [inline app/scripts/stores/layout.es6:17]: It's sort of different, and sort of the same. Here is what the JSX compiles down to with the object spread operator:

``` js
const otherProps = {a: 1, b: 2};
<Component prop="val" prop2="val2" {...otherProps} />
```

compiles to

``` js
React.createElement(Component, {
  prop: 'val',
  prop2: 'val2',
  ...otherProps
}, []);
```

so it's.... kind of... the same... ish

The nice thing about it is that it's a short way to create a new object with some things changed from the old one without mutating it. I'm going to pull in immutablejs though, which will enforce that and give us a more normal javascript syntax (and performance improvements and awesome stuff).

Comment (uniphil) [inline app/scripts/stores/layout.es6:17]: I will switch to immutablejs in a separate PR

**Comment by uniphil** [comment]
thanks!!

**Pull request merged**